src/common/libsubprocess/subprocess.c

```diff
@@ -246,11 +246,7 @@ void subprocess_standard_output (flux_subprocess_t *p, const char *stream)
 
 void subprocess_check_completed (flux_subprocess_t *p)
 {
-    if (p->state != FLUX_SUBPROCESS_EXITED) {
-        log_err ("subprocess_check_completed: unexpected state %s",
-                 flux_subprocess_state_string (p->state));
-        return;
-    }
+    assert (p->state == FLUX_SUBPROCESS_EXITED);
```

Comment:
I think it actually might be wise to not revert this commit.  Based on experience with el cap failures, we don't want to have this assert enabled.  So the question is probably do you want a log message when something weird happens that only happens at scale or do you want nothing at all when @grondo builds the RPM with NDEBUG.